tests/scripts/analyze_outcomes.py

```diff
@@ -95,32 +95,47 @@ def analyze_coverage(results, outcomes, allow_list, full_coverage):
             else:
                 results.warning('Allow listed test case was executed: {}', key)
 
+def name_matches_pattern(name, str_or_re):
+    """Check if name matches a pattern, that may be a string or regex.
+    - If the pattern is a string, name must be equal to match.
+    - If the pattern is a regex, name must fully match.
+    """
+    # The CI's python is too old for re.Pattern
+    #if isinstance(str_or_re, re.Pattern):
+    if not isinstance(str_or_re, str):
+        return str_or_re.fullmatch(name):
+    else:
+        return str_or_re == name:
```

Comment:
I think we want to distinguish between "exact string" and "regex" in `KNOWN_TASKS` - where I'll not all existing entries are "exact string" before this PR. The reason is that some of the "exact string" entries contain regex special chars like `.` or `(` which we do not want to be interpreted in a special way. For example if we have an entry for "Foo (bar)" then if we started interpreting it as a regex it would match "Foo bar" but no longer "Foo (bar)".

So, the primary reason for using `re.compile()` in `KNOW_TASKS` is to be able to have both kinds of entries ("exact string" and regex) in a single list and be able to distinguish them by type.

A secondary benefit is performance: the regexes are compiled only once.

Reply:
>The reason is that some of the "exact string" entries contain regex special chars like . or ( which we do not want to be interpreted in a special way.

Make sense to me, thanks!

Moving forward, please remove the colons at the end of line 106 and 108 :)

Reply:
> Moving forward, please remove the colons at the end of line 106 and 108 :)

:)